### PR TITLE
Alerting: Unify Swagger/OpenAPI generation tooling

### PR DESCRIPTION
**Pull request by alexweav**
**What this PR does / why we need it**:

This PR is one of a handful which improves the Swagger codegen tooling in the alerting API layer.

Currently, our Swagger codegen is a two-phase process. We first extract the Swagger doc from DTOs, using a makefile which is briefly documented. We then convert the extracted doc back into server stubs using an undocumented makefile in a separate directory. It is extremely rare for a developer to perform one step without the other. Neither makefile is invoked by CI tooling.

This PR unifies the two stages of generation into one single pass, to make it a bit more discoverable and obvious for other developers. This PR does not attempt to change any directory structures or file names in the API layer.

I'm submitting these changes in separate PRs so that the team can independently accept or decline them - the purpose of this change is to make adding new APIs easier for the team going forward.

### CODE REVIEW

File: pkg/services/ngalert/api/tooling/Makefile [hunk 1]
```diff
@@ -25,6 +29,30 @@ spec.json-mac: ensure_go-swagger_mac $(GO_PKG_FILES)
...
+fix:
+	sed -i -e 's/apimodels\.\[\]PostableAlert/apimodels.PostableAlerts/' $(GENERATED_GO_MATCHERS)
+	sed -i -e 's/apimodels\.\[\]UpdateDashboardAclCommand/apimodels.Permissions/' $(GENERATED_GO_MATCHERS)
+	sed -i -e 's/apimodels\.\[\]PostableApiReceiver/apimodels.TestReceiversConfigParams/' $(GENERATED_GO_MATCHERS)
+	goimports -w -v $(GENERATED_GO_MATCHERS)
+
+clean:
+	rm -rf ./go
+
+serve: post.json
```
Comment (alexweav) [inline pkg/services/ngalert/api/tooling/Makefile:55]: Previously, the tooling would start serving a [Swagger UI](https://swagger.io/tools/swagger-ui/) instance on port 80 as the last step. The makefile would always need to be Ctrl+C'd.

Instead, don't launch this by default. Put this functionality under `make serve` instead.

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM

**Comment by gotjosh** [comment]
@papagian can you please take a look?

**Review by papagian — APPROVED** [review]
it works as expected! 🎉

**Comment by JohnnyQQQQ** [comment]
@alexweav drone seems to be stuck, can you rebase or something to trigger it again.

**Pull request merged**